server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -213,11 +212,11 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
         }
       } else {
         LOG.info("compactionConfig is empty. Skip.");
-        updateAutoCompactionSnapshotWhenNoCompactTaskScheduled(currentRunAutoCompactionSnapshotBuilders);
+        autoCompactionSnapshotPerDataSource.set(new HashMap<>());
```

Comment:
just want to confirm. this will trigger when there is no compaction config for any datasource, right? If there are other datasources eligible for compaction and some are in-active, would this still be called?

Reply:
This will only trigger when you have no compaction config for any datasource

Reply:
In that case, where would we remove the status info for in-active data sources assuming some other datasources still have compaction enabled.

Reply:
autoCompactionSnapshotPerDataSource are built new from empty map every time the auto compaction runs. At the end of the auto compaction we do a atomic swap to replace the old map (which has in-active datasource) with the newly built map (which does not have in-active datasource)

Reply:
I see now. I was looking at the wrong change.  is this API used by the console to show the status info? I am wondering if it needs any changes on the console side to show the status for disabled datasources correctly.

Reply:
This API is used by the console to show the status info. The UI can already handle when one of the datasource does not have an entry in the map returned by this API. (This is for cases such as the datasource never has auto compaction enabled or auto compaction was previously enabled but is now disabled). Hence, no change in UI is needed.